planner.c

```diff
@@ -140,6 +141,7 @@ static void planner_recalculate()
 {   
   // Initialize block index to the last block in the planner buffer.
   uint8_t block_index = plan_prev_block_index(block_buffer_head);
+  magazine_gap_reset();
```

Comment:
I don't think this is the correct place to call this function. Ideally it should be called whenever settings.mag_gap_limit or settings.steps_per_mm[C_AXIS] changes. Currently we do not have the ability to update either of those on the fly without restarting the protocol main loop and calling magazine_init again, in which case the planner buffer will be reset in any case. Yes, eventually when we add the ability to change the values in the settings struct on the fly, we need to recalculate mag_state.delta_pos_limit amongst other things (such as waiting for the planner buffer to drain) which will include a much bigger rework. For now I think magazine_gap_reset can be merged with magazine_init.